### PR TITLE
Bump MSBuild.StructuredLogger to 2.1.820

### PR DESCRIPTION
**Pull request by gtbuchanan**
### Description

Updated with `paket update MSBuild.StructuredLogger --group fakemodule`

Resolves: #2744

## TODO

- [ ] New (API-)documentation for new features exist (Note: API-docs are enough, additional docs are in `help/markdown`)
- [ ] unit or integration test exists (or short reasoning why it doesn't make sense)
  
  > Note: Consider using the `CreateProcess` API which can be tested more easily, see https://github.com/fsharp/FAKE/pull/2131/files#diff-4fb4a77e110fbbe8210205dfe022389b for an example (the changes in the `DotNet.Testing.NUnit` module)
  
- [x] boy scout rule: "leave the code behind in a better state than you found it" (fix warnings, obsolete members or code-style in the places you worked in)     
- [ ] (if new module) the module has been linked from the "side navigation" menu, edit `docs/data.json`.
- [ ] (if new module) the module is in the correct namespace.
- [ ] (if new module) the module is added to Fake.sln (`dotnet sln Fake.sln add src/app/Fake.*/Fake.*.fsproj`)
- [x] Fake [API guideline](https://fake.build/guide/contributing.html#API-Design-Guidelines) is honored

### CODE REVIEW

File: paket.lock [hunk 1]
```diff
@@ -794,19 +799,17 @@ NUGET
...
-    Microsoft.Win32.SystemEvents (6.0) - restriction: || (== net6.0) (&& (== netcoreapp2.1) (>= netcoreapp3.1)) (&& (== netstandard2.0) (>= netcoreapp3.1))
+    Microsoft.Win32.SystemEvents (6.0) - restriction: || (&& (== net472) (>= net7.0)) (== net6.0) (&& (== netcoreapp2.1) (>= netcoreapp3.1)) (&& (== netstandard2.0) (>= netcoreapp3.1))
     Mono.Cecil (0.11.4)
     Mono.Posix.NETStandard (1.0)
-    MSBuild.StructuredLogger (2.1.545)
-      Microsoft.Build (>= 16.10)
-      Microsoft.Build.Framework (>= 16.10)
-      Microsoft.Build.Tasks.Core (>= 16.10)
-      Microsoft.Build.Utilities.Core (>= 16.10)
+    MSBuild.StructuredLogger (2.1.820)
```
Comment (cataggar) [inline paket.lock:810]: 2.1.545 August 2021 to 2.1.820  May 2023. https://www.nuget.org/packages/MSBuild.StructuredLogger

**Review by cataggar — APPROVED** [review]

File: paket.lock (hunk 1 above)
Comment (xperiandri) [reply paket.lock:810]: Can we make it wildcard? `2.1.*`

Comment (xperiandri) [reply paket.lock:810]: Looks like it changes pretty often

Comment (xperiandri) [reply paket.lock:810]: Should be `(>= 2.1.0)` right?

**Comment by xperiandri** [comment]
@gtbuchanan I'm a new maintainer. Do you want to finish your PR?

**Comment by Numpsy** [comment]
So what needs to be done to get this finished?
(There are a few more versions post the one updated to here, and the release notes for the latest one says that it adds support for another new format version, so maybe it should be jumped up to that version)

**Comment by Numpsy** [comment]
An observation:
paket.dependencies currently contains
```
nuget MSBuild.StructuredLogger
        nuget Microsoft.Build
        nuget Microsoft.Build.Framework
        nuget Microsoft.Build.Tasks.Core
        nuget Microsoft.Build.Utilities.Core
```
and those sub packages match the deps of the 2.1.545 version currently in use:
image

However, the current structured loggers versions have reduced deps of 
image

so does paket.dependencies also need changing to match?

**Comment by Numpsy** [comment]
> maybe it should be jumped up to that version

Or not, seems that updating the microsoft.build deps to 17.5 causes the integration tests to fail with the error described in https://github.com/fsprojects/FAKE/issues/2722, so maybe leaving the min version at 2.1.815 that works with the older MSBuild versions would be safest (that should still support the v16 binlog format)

**Comment by xperiandri** [comment]
@Numpsy OK, will you submit a PR?

**Pull request closed without merging**